pkg/alertmanager/multitenant.go

```diff
@@ -270,33 +270,24 @@ func (am *MultitenantAlertmanager) updateConfigs(now time.Time) error {
 }
 
 // poll the configuration server. Not re-entrant.
-func (am *MultitenantAlertmanager) poll() (map[string]configs.View, error) {
-	configID := am.latestConfig
-	cfgs, err := am.configsAPI.GetAlerts(context.Background(), configID)
+func (am *MultitenantAlertmanager) poll() (map[string]alerts.AlertConfigDesc, error) {
```

Comment:
Given `alerts.AlertStore` is an interface easily mockable, would make much sense add unit tests to cover at least a basic scenario:

- `loadAllConfigs()` with 2 users (A and B)
- Assert the state is as expected
- Add user C and remove user B
- `poll()`
- Assert the state is as expected (C added, B removed)

Reply:
+1